src/Connection.ts

```diff
@@ -136,7 +149,10 @@ class Connection {
 		cacheMaxAge: number,
 		/** Request timeout (ms) */
 		timeout: number,
+		options: ConnectionConstructorOptions,
 	) {
+		const { httpAgent, httpsAgent } = options;
```

Comment:
So _someone_ has to worry about destroying them I'm thinking.  There is no facility in mvom to destroy a connection other than letting the connection object drop from scope, and to my knowledge, there's no way to tap into an object's lifecycle to run anything which could destroy the http agent.  Since the consumers are providing the agent and are responsible for the creation of the agent, the burden for destroying the agent will have to fall on them as well.